doctr/__main__.py

```diff
@@ -47,25 +60,57 @@ def main():
 
     if on_travis:
         repo = get_repo()
-        if setup_GitHub_push(repo):
+        if setup_GitHub_push(repo, auth_type='token' if args.token else 'deploy_key'):
             commit_docs()
             push_docs()
     else:
-        username = input("What is your GitHub username? ")
-        token = generate_GitHub_token(username)
-
         repo = input("What repo to you want to build the docs for? ")
-        encrypted_variable = encrypt_variable("GH_TOKEN={token}".format(token=token).encode('utf-8'), repo=repo)
-        travis_content = """
-env:
-  global:
-    secure: "{encrypted_variable}"
 
-""".format(encrypted_variable=encrypted_variable.decode('utf-8'))
+        if args.token:
+            token = generate_GitHub_token()
+            encrypted_variable = encrypt_variable("GH_TOKEN={token}".format(token=token).encode('utf-8'), repo=repo)
+        else:
+            ssh_key = generate_ssh_key("doctr deploy key for {repo}".format(repo=repo))
+            key = encrypt_file('github_deploy_key', delete=True)
+            encrypted_variable = encrypt_variable(b"DOCTR_DEPLOY_ENCRYPTION_KEY=" + key, repo=repo)
+
+            deploy_keys_url = 'https://github.com/{repo}/settings/keys'.format(repo=repo)
+
+            if args.upload_key:
+
+                upload_GitHub_deploy_key(repo, ssh_key)
+
+                print(dedent("""\
+                The deploy key has been added for {repo}.
+
+                Commit the file github_deploy_key.enc to the repository.
+
+                You can go to {deploy_keys_url} to revoke the deploy key.
+                """.format(repo=repo, deploy_keys_url=deploy_keys_url)))
+
+            else:
+                print(dedent("""\
+                Go to {deploy_keys_url} and add the following as a new key:
+
+                {ssh_key}
+
+                Be sure to allow write access for the key.
+                """.format(ssh_key=ssh_key, deploy_keys_url=deploy_keys_url)))
+
+            # TODO: Should we just delete the public key?
+
+            print(dedent("""Commit the file github_deploy_key.enc. The file
+            github_deploy_key.pub contains the public deploy key for GitHub.
+            It does not need to be committed."""))
```

Comment:
Should we just delete the public key? It's either uploaded to GitHub or printed to the screen. It's harmless if committed, but unimportant.